octavia-cli/octavia_cli/generate/renderer.py

```diff
@@ -0,0 +1,160 @@
+
+from .definitions import BaseDefinition
+
+JINJA_ENV = Environment(loader=PackageLoader("octavia_cli"), autoescape=select_autoescape(), trim_blocks=False, lstrip_blocks=True)
+
+
+class FieldToRender:
+    def __init__(self, name: str, required: bool, field_metadata: dict) -> None:
+        """Initialize a FieldToRender instance
+
+        Args:
+            name (str): name of the field
+            required (bool): whether it's a required field or not
+            field_metadata (dict): metadata associated with the field
+        """
+        self.name = name
+        self.required = required
+        self.field_metadata = field_metadata
+        self.one_of_values = self._get_one_of_values()
+        self.object_properties = get_object_fields(field_metadata)
+        self.array_items = self._get_array_items()
+        self.comment = self._build_comment(
+            [
+                self._get_secret_comment,
+                self._get_required_comment,
+                self._get_type_comment,
+                self._get_description_comment,
+                self._get_example_comment,
+            ]
+        )
+        self.default = self._get_default()
+
+    def __getattr__(self, name: str) -> Any:
+        """Map field_metadata keys to attributes of Field.
+
+        Args:
+            name (str): attribute name
+
+        Returns:
+            [Any]: attribute value
+        """
+        if name in self.field_metadata:
+            return self.field_metadata.get(name)
+
+    @property
+    def is_array_of_objects(self) -> bool:
+        if self.type == "array" and self.items:
+            if self.items["type"] == "object":
+                return True
+        return False
+
+    def _get_one_of_values(self) -> List[List["FieldToRender"]]:
+        """An object field can have multiple kind of values if it's a oneOf.
+        This functions returns all the possible one of values the field can take.
+        Returns:
+            [list]: List of oneof values.
+        """
+        if not self.oneOf:
+            return []
+        one_of_values = []
+        for one_of_value in self.oneOf:
+            properties = get_object_fields(one_of_value)
+            one_of_values.append(properties)
+        return one_of_values
+
+    def _get_array_items(self) -> List["FieldToRender"]:
+        """If the field is an array of objects, retrieve fields of these objects.
+
+        Returns:
+            [list]: List of fields
+        """
+        if self.is_array_of_objects:
+            required_fields = self.items.get("required", [])
+            return parse_fields(required_fields, self.items["properties"])
+        return []
+
+    def _get_required_comment(self) -> str:
+        return "REQUIRED" if self.required else "OPTIONAL"
+
+    def _get_type_comment(self) -> str:
+        return self.type if self.type else None
+
+    def _get_secret_comment(self) -> str:
+        return "🤫" if self.airbyte_secret else None
```

Comment:
Do you think an emoji here or something more simple like `SECRET` would be more clear to users? When putting myself in the perspective of a potential user, I feel like I would be confused at what the emoji means

Reply:
You're 💯 right 😄 . In an upcoming PR, we're going to split the YAML file in two: one file for regular configs, another part with secrets only that will be git ignored. But I'm going to change this emoji now.

Reply:
I made the change.